platform/android/app/src/androidTest/kotlin/coop/polypoly/polypod/FeatureTest.kt

```diff
@@ -0,0 +1,81 @@
+package coop.polypoly.polypod
+
+import androidx.compose.ui.test.junit4.AndroidComposeTestRule
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performScrollTo
+import androidx.preference.PreferenceManager
+import androidx.test.espresso.web.sugar.Web.onWebView
+import androidx.test.espresso.web.webdriver.DriverAtoms.* // ktlint-disable no-wildcard-imports
+import androidx.test.espresso.web.webdriver.Locator
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import coop.polypoly.core.FeatureCategoryId
+import coop.polypoly.polypod.core.UpdateNotification
+import coop.polypoly.polypod.features.FeatureStorage
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+private typealias MainTestRule =
+    AndroidComposeTestRule<ActivityScenarioRule<MainActivity>, MainActivity>
+
+private fun waitFor(timeout: Long, f: () -> Unit) {
+    val interval = 100L
+    val startTime = System.currentTimeMillis()
+    while (true) {
+        Thread.sleep(interval)
+        try {
+            f()
+            break
+        } catch (e: Exception) {
+            if (System.currentTimeMillis() - startTime > timeout)
+                throw RuntimeException("Timeout after ${timeout}ms", e)
+        }
+    }
+}
+
+@RunWith(AndroidJUnit4::class)
+class FeatureTest {
+    @get:Rule
+    val composeTestRule: MainTestRule by lazy {
+        val context = InstrumentationRegistry.getInstrumentation().targetContext
+        PreferenceManager.getDefaultSharedPreferences(context).edit().clear()
+            .commit()
+        Preferences.setFirstRun(context, false)
+        Preferences.setSecurityDoNotAskAgainCheck(context, true)
+        UpdateNotification.mockData.id = 0
+        FeatureStorage.forceShowCategories = listOf(FeatureCategoryId.developer)
+        createAndroidComposeRule()
+    }
+
+    @Test
+    fun runAllTestFeatureTests() {
+        composeTestRule.onNodeWithText("Test").run {
+            performScrollTo()
+            performClick()
+        }
+
+        onFeature()
+            .withElement(findElement(Locator.ID, "runAll"))
+            .perform(webClick())
+
+        val successStatus = "All OK"
+        try {
+            waitFor(10000) {
```

Comment:
We might need to make this bigger when we add more tests to the test feature right?

Reply:
Yeah, probably, but I tried to make sure we get a very clear stack trace :D I think in the future, we need to rewrite this to run each test individually rather than `runAll`.

Reply:
It is a good start though!